docs/hazmat/primitives/asymmetric/dh.rst

```diff
@@ -19,7 +19,9 @@ Exchange Algorithm
 ~~~~~~~~~~~~~~~~~~
 
 For most applications the ``shared_key`` should be passed to a key
-derivation function.
+derivation function in order to produce a key with the expected number of
+random bits, and to destroy mathematical structure in the shared key which
```

Comment:
After soliciting several people for ideas on phrasing here how about:

"...should be passed to a key derivation function. This allows mixing of additional information into the key, derivation of multiple keys, and destroys any structure that may be present."

Not really much better I guess ;)